### PR TITLE
update docgen documentation

### PR DESCRIPTION
**Pull request by a-mr**
Mentions `doc2tex` in Changelog and `docgen.rst` (for #17997 and #18141).

Fix a few typos/anachronisms in `docgen.rst`.

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -449,6 +449,13 @@
 
 ## Tool changes
 
+- Latex doc generation is revised: output `.tex` files should be compiled
+  by `xelatex` (not by `pdflatex` as before). Now default Latex settings
+  provide support for Unicode and do better job for avoidind margin overflows.
```
Comment (timotheecour) [inline changelog.md]: avoidind

(you can use `[skip ci] fix changelog` in commit msg)

File: doc/docgen.rst [hunk 2]
```diff
@@ -14,12 +14,22 @@ Introduction
...
 (reStructuredText) files. The output documentation will include the module
 dependencies (`import`), any top-level documentation comments (`##`), and
 exported symbols (`*`), including procedures, types, and variables.
 
+===================   ======================   ============   ==============
+command               runs on...               input format   output format
+===================   ======================   ============   ==============
+`nim doc`:cmd:        documentation comments   ``.nim``       ``.html`` HTML
+`nim doc2tex`:cmd:    \`                       \`             ``.tex`` LaTeX
+`nim jsondoc`:cmd:    \`                       \`             ``.json`` JSON
```
Comment (timotheecour) [inline doc/docgen.rst]: how about (in this or future PR) change this to doc2json for consistency (and making jsondoc an undocumented alias to it)

File: doc/docgen.rst [hunk 3]
```diff
@@ -14,12 +14,22 @@ Introduction
...
+`nim rst2html`:cmd:   standalone documents     ``.rst``       ``.html`` HTML
```
Comment (timotheecour) [inline doc/docgen.rst]:
```suggestion
`nim rst2html`:cmd:   standalone rst files     ``.rst``       ``.html`` HTML
```

File: doc/docgen.rst [hunk 4]
```diff
@@ -14,12 +14,22 @@ Introduction
...
+from input ``.nim`` files and projects, as well as HTML and LaTeX from input RST
 (reStructuredText) files. The output documentation will include the module
 dependencies (`import`), any top-level documentation comments (`##`), and
 exported symbols (`*`), including procedures, types, and variables.
 
+===================   ======================   ============   ==============
+command               runs on...               input format   output format
+===================   ======================   ============   ==============
+`nim doc`:cmd:        documentation comments   ``.nim``       ``.html`` HTML
+`nim doc2tex`:cmd:    \`                       \`             ``.tex`` LaTeX
```
Comment (timotheecour) [inline doc/docgen.rst]: (nit) how about `-` instead of backtick ?

File: doc/docgen.rst [hunk 5]
```diff
@@ -99,32 +110,22 @@ won't influence RST formatting.
...
 
-  helloWorld(5)
-
-
-Document Types
-==============
+.. code:: nim
+   :file: docgen_sample.nim
 
+All the below commands save their output to ``htmldocs`` directory relatieve to
```
Comment (timotheecour) [inline doc/docgen.rst]: relatieve

File: doc/docgen.rst [hunk 6]
```diff
@@ -217,9 +239,9 @@ Project switch
 
   nim doc --project filename.nim
 
-This will recursively generate documentation of all nim modules imported
+This will recursively generate documentation of all Nim modules imported
 into the input module that belong to the Nimble package that ``filename.nim``
-belongs to.
+belongs to. The index files will also be generated.
```
Comment (timotheecour) [inline doc/docgen.rst]: also mention `theindex.html`

File: doc/docgen.rst (hunk 4 above)
Comment (a-mr) [reply doc/docgen.rst]: I don't know what `-` means. While \` aka "ditto" means "the same". I just don't want repeat "documentation comments" 3 times.

File: doc/docgen.rst (hunk 2 above)
Comment (a-mr) [reply doc/docgen.rst]: but if you push this logic than `doc2html` shall be introduced instead of `doc`.

File: doc/docgen.rst (hunk 4 above)
Comment (timotheecour) [reply doc/docgen.rst]: i didn't know about backtick had meaning of ditto, is that standard? anyways, I don't care too much either way

File: doc/docgen.rst (hunk 6 above)
Comment (a-mr) [reply doc/docgen.rst]: ok, thx.

File: doc/docgen.rst (hunk 5 above)
Comment (a-mr) [reply doc/docgen.rst]: yes.

File: doc/docgen.rst (hunk 4 above)
Comment (a-mr) [reply doc/docgen.rst]: it seems I confused something. Replaced to proper unicode ditto.

File: changelog.md (hunk 1 above)
Comment (a-mr) [reply changelog.md]: thanks.

**Pull request merged**